tests/unit/s2n_x509_validator_test_offset.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+static int mock_time(void *data, uint64_t *timestamp)
+{
+    *timestamp = *(uint64_t *) data;
+    return 0;
+}
+
+static int read_file(struct s2n_stuffer *file_output, const char *path, uint32_t max_len)
+{
+    FILE *fd = fopen(path, "rb");
+    s2n_stuffer_alloc(file_output, max_len);
+
+    if (fd) {
+        char data[1024];
+        size_t r = 0;
+        while ((r = fread(data, 1, sizeof(data), fd)) > 0) {
+            s2n_stuffer_write_bytes(file_output, (const uint8_t *) data, (const uint32_t) r);
+        }
+        fclose(fd);
+        return s2n_stuffer_data_available(file_output) > 0;
+    }
+
+    return -1;
+}
+
+struct host_verify_data {
+    const char *name;
+    uint8_t found_name;
+    uint8_t callback_invoked;
+};
+
+static uint8_t verify_host_accept_everything(const char *host_name, size_t host_name_len, void *data)
+{
+    return 1;
+}
+
+int test_with_system_time(uint64_t system_time_nanos, int expected)
+{
+    DEFER_CLEANUP(struct s2n_x509_trust_store trust_store = { 0 }, s2n_x509_trust_store_wipe);
```

Comment:
nit: `trust_store` isn't a pointer, so I'm not sure it matters if it's cleaned up or not when it goes out of scope.

Reply:
I was burned by my OpenSSL experience where my 0 initialized struct occasionally allocated data 😅 . Do you know why we call `s2n_x509_trust_store_wipe` for all of the test cases in the `s2n_x509_validator_test`?